core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -563,6 +565,18 @@ class LogManager(logDirs: Seq[File],
     startupWithConfigOverrides(defaultConfig, fetchTopicConfigOverrides(defaultConfig, topicNames))
   }
 
+  def deleteStrayKRaftReplicas(
+    brokerId: Int,
+    image: TopicsImage
+  ): Unit = {
+    val strayPartitions = findStrayReplicas(brokerId, image, allLogs)
+    strayPartitions.foreach(topicPartition => {
+      val log = getLog(topicPartition).get
+      log.renameDir(UnifiedLog.logStrayDirName(topicPartition), shouldReinitialize = true)
+      asyncDelete(topicPartition, false, false, true)
```

Comment:
1. We should not `Reinitialize` the metadata since the stray partitions are unused anymore.
2. The renameDir is already done in `asyncDelete` if `isStray` is true, which is what we have now. So we don't need to renameDir here.

Reply:
OK. I will delete the call to `log.renameDir`.

Reply:
I didn't see this change. Did you miss it?